src/partisan_hiscamp_membership_strategy.erl

```diff
@@ -61,14 +61,23 @@ join(#hiScamp{membership=Membership0, level1=L1, level2=L2}=State0, Node, _NodeS
     MembershipList = membership_list(State),
     {ok, Membership, OutgoingMessages, State}.
 
-distance(#hiScamp{membership=Membership0, heap=Heap, level1=L1, level2=L2}=_State0, Node) ->
-    % A node j joins the system by sending a subscription request to the node s which is closest to it
-    Threshold = 3,
+try_distance(#hiScamp{membership=Membership0, distance=Dist}=_State0, Node) ->
     Dict = get(distance_metrics),
     try dict:find(Node, Dict) of
```

Comment:
Find returns either ```{ok, Value}``` or ```false```.  It won't throw an exception; it returns an error, so you shouldn't be using a try statement, you should be using a case statement to match against the possible return values.